serviceregistry-server/src/main/java/no/difi/meldingsutveksling/serviceregistry/controller/ServiceRecordController.java

```diff
@@ -42,159 +52,160 @@
-
-        String orgnr = entityInfo.getIdentifier();
-        if (shouldCreateServiceRecordForCititzen().test(entityInfo)) {
-            Optional<ServiceRecord> serviceRecord = Optional.empty();
-            try {
-                serviceRecord = serviceRecordFactory.createServiceRecordForCititzen(orgnr, auth, clientOrgnr, obligation, forcePrint);
-            } catch (KRRClientException e) {
-                log.error("Error looking up identifier in KRR", e);
-            }
-            serviceRecord.ifPresent(r -> entity.getServiceRecords().add(r));
-        }
-
-        Optional<ServiceRecord> eduServiceRecord = serviceRecordResponseHandler(serviceRecordFactory.createEduServiceRecord(orgnr), entity);
-        eduServiceRecord.ifPresent(r -> entity.getServiceRecords().add(r));
-
-        Optional<FiksWrapper> fiksWrapper = serviceRecordFactory.createFiksServiceRecord(orgnr);
-        if (fiksWrapper.isPresent()) {
-            Optional<ServiceRecord> fiksServiceRecord = serviceRecordResponseHandler(Optional.of(fiksWrapper.get().getServiceRecord()), entity);
-            if (fiksServiceRecord.isPresent()) {
-                entity.getServiceRecords().add(fiksServiceRecord.get());
-                entity.getSecuritylevels().put(ServiceIdentifier.DPF, fiksWrapper.get().getSecuritylevel());
-            }
-        }
-
-        Optional<ServiceRecord> dpeInnsynServiceRecord = serviceRecordResponseHandler(serviceRecordFactory.createDpeInnsynServiceRecord(orgnr), entity);
-        dpeInnsynServiceRecord.ifPresent(r -> entity.getServiceRecords().add(r));
-
-        Optional<ServiceRecord> dpeDataServiceRecord = serviceRecordResponseHandler(serviceRecordFactory.createDpeDataServiceRecord(orgnr), entity);
-        dpeDataServiceRecord.ifPresent(r -> entity.getServiceRecords().add(r));
-
-        if (dpeInnsynServiceRecord.isPresent() || dpeDataServiceRecord.isPresent()) {
-            Optional<ServiceRecord> dpeReceiptServiceRecord = serviceRecordResponseHandler(serviceRecordFactory.createDpeReceiptServiceRecord(orgnr), entity);
-            dpeReceiptServiceRecord.ifPresent(r -> entity.getServiceRecords().add(r));
-        }
-
-        Optional<ServiceRecord> dpvServiceRecord = serviceRecordFactory.createPostVirksomhetServiceRecord(orgnr);
-        dpvServiceRecord.ifPresent(r -> entity.getServiceRecords().add(r));
+    private ResponseEntity errorResponse(HttpStatus status, String message) {
+        return ResponseEntity.status(status)
+                .body(ErrorResponse.builder().errorDescription(message).build());
+    }
 
+    private ResponseEntity notFoundResponse(String logMessage) {
+        log.error(markerFrom(requestScope), logMessage);
+        return ResponseEntity.notFound().build();
     }
 
     @RequestMapping(value = "/identifier/{identifier}", method = RequestMethod.GET, produces = "application/jose")
     @ResponseBody
     public ResponseEntity signed(
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Refactor this method to throw at most one checked exception instead of: no.difi.meldingsutveksling.serviceregistry.security.EntitySignerException, no.difi.meldingsutveksling.serviceregistry.exceptions.SecurityLevelNotFoundException, no.difi.meldingsutveksling.serviceregistry.krr.KRRClientException, no.difi.meldingsutveksling.serviceregistry.CertificateNotFoundException, no.difi.meldingsutveksling.serviceregistry.krr.DsfLookupException, no.difi.meldingsutveksling.serviceregistry.service.brreg.BrregNotFoundException [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar-app01.difi.local:9000/coding_rules#rule_key=squid%3AS1160)